### PR TITLE
Add SNAP immigration status eligibility

### LINKED ISSUE #1475: SNAP immigration-related rules
https://www.fns.usda.gov/snap/eligibility/citizen/non-citizen-policy

### PR DESCRIPTION
**Pull request by hua7450**
## Summary

Closes #1475

Adds SNAP immigration status eligibility, reflecting changes from the One Big Beautiful Bill Act of 2025 (OBBBA). Excludes immigration-ineligible members from SNAP unit size for correct benefit calculation in mixed-status households. Implements California's delayed effective date (April 1, 2026) per ACL 25-92.

## Regulatory Authority

- [7 USC 2015(f)](https://www.law.cornell.edu/uscode/text/7/2015#f) - Eligibility disqualifications
- [8 USC 1612](https://www.law.cornell.edu/uscode/text/8/1612) - Limited eligibility of qualified aliens for certain Federal programs
- [USDA FNS - SNAP Implementation of OBBB](https://www.fns.usda.gov/snap/obbb-alien-eligibility) - Alien SNAP Eligibility guidance
- [7 CFR 273.11(c)](https://www.law.cornell.edu/cfr/text/7/273.11) - Treatment of income and resources of excluded household members
- [California CDSS ACL 25-92](https://www.cdss.ca.gov/Portals/9/Additional-Resources/Letters-and-Notices/ACLs/2025/25-92.pdf#page=7) - CalFresh implementation of H.R. 1 noncitizen eligibility changes

### Parameters
- `gov/usda/snap/eligibility/eligible_immigration_statuses.yaml`
  - Pre-OBBBA (2018-01-01): CITIZEN, LPR, REFUGEE, ASYLEE, DEPORTATION_WITHHELD, CUBAN_HAITIAN_ENTRANT, CONDITIONAL_ENTRANT, PAROLED_ONE_YEAR
  - Post-OBBBA (2025-07-04): CITIZEN, LPR, CUBAN_HAITIAN_ENTRANT only
- `gov/states/ca/cdss/snap/eligibility/eligible_immigration_statuses.yaml`
  - Same pre-OBBBA list, but restricted list effective **2026-04-01** (California delayed implementation per ACL 25-92)

### Variables
- `is_snap_immigration_status_eligible` - Person-level variable checking if immigration status is eligible; uses `federal_eligible | ca_eligible` to apply CA's delayed date
- `ca_snap_immigration_status_eligible` - California-specific immigration eligibility with `defined_for = StateCode.CA`
- Updated `is_snap_eligible` to require at least one household member with eligible immigration status
- Updated `snap_unit_size` (renamed from `snap_unit.py`) to exclude immigration-ineligible members, using OR with ineligible students to prevent double-counting

### Mixed-Status Households
- `is_snap_eligible`: Uses `spm_unit.any()` — household is eligible if at least one member has qualifying status ([7 CFR 273.11(c)](https://www.law.cornell.edu/cfr/text/7/273.11))
- `snap_unit_size`: Subtracts ineligible members so allotment, standard deduction, and min allotment are based on eligible member count only
- Income proration for ineligible aliens is not yet implemented (separate follow-up)

## Known Limitations

- **COFA citizens**: Eligible post-OBBBA per federal law, but `COFA` is not in the `ImmigrationStatus` enum. Needs to be added separately.
- **LPR 5-year waiting period**: Cannot be modeled in PolicyEngine's single-period architecture.
- **Income proration**: Per [7 CFR 273.11(c)](https://www.law.cornell.edu/cfr/text/7/273.11), ineligible aliens' income should be prorated. Not yet implemented.

## Test Plan
- [x] `is_snap_immigration_status_eligible` - 10 tests covering federal and CA-specific behavior
- [x] `ca_snap_immigration_status_eligible` - 8 tests covering CA delayed implementation
- [x] `snap_unit_size` - 9 tests including mixed-status households and double-counting prevention
- [x] `is_snap_eligible` - 8 tests including undocumented ineligible student edge case
- [x] Existing SNAP eligibility tests pass

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: policyengine_us/variables/gov/usda/snap/eligibility/is_snap_eligible.py [hunk 1]
```diff
@@ -28,8 +28,12 @@ def formula(spm_unit, period, parameters):
         work_requirements_eligibility = spm_unit(
             "meets_snap_work_requirements", period
         )
+        immigration_eligible_member_present = spm_unit.any(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/usda/snap/eligibility/is_snap_eligible.py:32]: it's not calculated like students with prorating?

Comment (hua7450) [reply policyengine_us/variables/gov/usda/snap/eligibility/is_snap_eligible.py:32]: I haven't check, this draft is 100% out of claude code. 
But yes, should consider the prorating thing.
